### PR TITLE
JAMES-2352 use @Override instead of javadoc comments

### PR DESCRIPTION
**Pull request by mbaechler**

### CODE REVIEW

File: server/container/util/src/main/java/org/apache/james/util/retry/ExceptionRetryHandler.java [hunk 1]
```diff
@@ -72,6 +72,7 @@ public ExceptionRetryHandler(Class<?>[] exceptionClasses, ExceptionRetryingProxy
 
         /**
          */
```
Comment (rouazana) [inline server/container/util/src/main/java/org/apache/james/util/retry/ExceptionRetryHandler.java]: remove useless comments in this file?

File: server/container/util/src/test/java/org/apache/james/util/retry/naming/NamingExceptionRetryHandlerTest.java [hunk 2]
```diff
@@ -52,31 +52,23 @@ public void setUp() throws Exception {
...
     public Context newDelegate() throws NamingException {
         return null;
     }
 
-    /**
-     */
     @Override
     public void resetDelegate() throws NamingException {
     }
     }
```
Comment (rouazana) [inline server/container/util/src/test/java/org/apache/james/util/retry/naming/NamingExceptionRetryHandlerTest.java]: indent?

File: server/mailet/mailetcontainer-camel/src/main/java/org/apache/james/mailetcontainer/lib/AbstractStateMailetProcessor.java [hunk 3]
```diff
@@ -399,6 +396,7 @@ private void parseConfiguration() throws MessagingException, ConfigurationExcept
          * @see
          * org.apache.mailet.base.GenericMailet#service(org.apache.mailet.Mail)
          */
```
Comment (rouazana) [inline server/mailet/mailetcontainer-camel/src/main/java/org/apache/james/mailetcontainer/lib/AbstractStateMailetProcessor.java]: remove comment?

File: mailbox/jpa/src/main/java/org/apache/james/mailbox/jpa/mail/JPAMessageMapper.java [hunk 4]
```diff
@@ -299,10 +300,6 @@ private MessageMetaData copy(Mailbox mailbox, MessageUid uid, long modSeq, Mailb
         return save(mailbox, copy);
     }
 
-    /**
-     * @see org.apache.james.mailbox.store.mail.AbstractMessageMapper#save(Mailbox,
-     *      MailboxMessage)
-     */
```
Comment (rouazana) [inline mailbox/jpa/src/main/java/org/apache/james/mailbox/jpa/mail/JPAMessageMapper.java]: missing @override?

File: server/container/util/src/test/java/org/apache/james/util/retry/naming/NamingExceptionRetryHandlerTest.java [hunk 5]
```diff
@@ -40,84 +40,81 @@
     private ExceptionRetryingProxy proxy = null;
     private RetrySchedule schedule = null;
 
-    /**
-     * @see junit.framework.TestCase#setUp()
-     */
     @Before
```
Comment (rouazana) [inline server/container/util/src/test/java/org/apache/james/util/retry/naming/NamingExceptionRetryHandlerTest.java]: missing @override?

File: server/container/util/src/test/java/org/apache/james/util/retry/naming/NamingExceptionRetryHandlerTest.java [hunk 6]
```diff
@@ -40,84 +40,81 @@
...
-    }
+        @Override
+        public long getInterval(int index) {
+            return index;
+        }
     }
 
     /**
      * Test method for .
      */
```
Comment (rouazana) [inline server/container/util/src/test/java/org/apache/james/util/retry/naming/NamingExceptionRetryHandlerTest.java]: these comments can be removed too

File: server/container/util/src/test/java/org/apache/james/util/retry/naming/NamingExceptionRetryHandlerTest.java [hunk 7]
```diff
@@ -31,24 +31,9 @@
 import org.junit.Before;
 import org.junit.Test;
 
-/**
```
Comment (rouazana) [inline server/container/util/src/test/java/org/apache/james/util/retry/naming/NamingExceptionRetryHandlerTest.java]: this commit 7ee73d3d99f4e45fe793a6b23b071ad45d41555b does more than just use @override, it should be separated

File: mailbox/store/src/main/java/org/apache/james/mailbox/store/SimpleMailboxMetaData.java [hunk 8]
```diff
@@ -45,7 +45,6 @@ public SimpleMailboxMetaData(MailboxPath path, MailboxId mailboxId, char delimit
     }
 
     public SimpleMailboxMetaData(MailboxPath path, MailboxId mailboxId, char delimiter, Children inferiors, Selectability selectability) {
-        super();
```
Comment (rouazana) [inline mailbox/store/src/main/java/org/apache/james/mailbox/store/SimpleMailboxMetaData.java]: why? + unrelated to this commit. Really, please try to not mix big commits with (even little) changes

File: mailet/mailetdocs-maven-plugin/src/main/java/org/apache/james/mailet/MailetMatcherDescriptor.java [hunk 9]
```diff
@@ -50,9 +50,6 @@ public String getName() {
             return name;
         }
 
-        /**
-         * @see #getName()
-         */
```
Comment (rouazana) [inline mailet/mailetdocs-maven-plugin/src/main/java/org/apache/james/mailet/MailetMatcherDescriptor.java]: missing @override?

File: protocols/api/src/main/java/org/apache/james/protocols/api/Encryption.java [hunk 10]
```diff
@@ -37,10 +37,6 @@ private Encryption(SSLContext context, boolean starttls, String[] enabledCipherS
         this.enabledCipherSuites = enabledCipherSuites;
     }
 
-    /**
-     * 
-     * @see #createStartTls(SSLContext, String[])
-     */
```
Comment (rouazana) [inline protocols/api/src/main/java/org/apache/james/protocols/api/Encryption.java]: missing @override?

File: protocols/api/src/main/java/org/apache/james/protocols/api/Encryption.java [hunk 11]
```diff
@@ -58,10 +54,6 @@ public static Encryption createTls(SSLContext context, String[] enabledCipherSui
         return new Encryption(context, false, enabledCipherSuites);
     }
 
-    /**
-     * 
-     * @see #createStartTls(SSLContext, String[])
-     */
```
Comment (rouazana) [inline protocols/api/src/main/java/org/apache/james/protocols/api/Encryption.java]: missing @override?

File: server/container/util/src/test/java/org/apache/james/util/retry/ExceptionRetryHandlerTest.java [hunk 12]
```diff
@@ -33,154 +33,122 @@
 import org.junit.Before;
 import org.junit.Test;
 
-/**
- * <code>ExceptionRetryHandlerTest</code>
- */
 public class ExceptionRetryHandlerTest {
+    private static class TestRetryingProxy implements ExceptionRetryingProxy {
```
Comment (rouazana) [inline server/container/util/src/test/java/org/apache/james/util/retry/ExceptionRetryHandlerTest.java]: unrelated change

File: server/data/data-file/src/main/java/org/apache/james/sieverepository/file/SieveFileRepository.java [hunk 13]
```diff
@@ -168,14 +168,6 @@ public InputStream getScript(String user, String name) throws ScriptNotFoundExce
         return script;
     }
 
-    /**
-     * The default quota, if any, is stored in file '.quota' in the sieve root directory. Quotas for
-     * specific users are stored in file '.quota' in the user's directory.
-     * <p/>
-     * <p>The '.quota' file contains a single positive integer value representing the quota in octets.
```
Comment (rouazana) [inline server/data/data-file/src/main/java/org/apache/james/sieverepository/file/SieveFileRepository.java]: this doc is not in the parent class

File: server/data/data-file/src/main/java/org/apache/james/user/file/UsersFileRepository.java [hunk 14]
```diff
@@ -198,10 +198,7 @@ public boolean contains(String name) throws UsersRepositoryException {
         }
     }
 
-    /*
-     * This is not longer in the api (deprecated)
-     * @see org.apache.james.user.api.UsersRepository#containsCaseInsensitive(java.lang.String)
-     */
+    @Deprecated
```
Comment (rouazana) [inline server/data/data-file/src/main/java/org/apache/james/user/file/UsersFileRepository.java]: unrelated commit

File: server/data/data-ldap/src/main/java/org/apache/james/user/ldap/ReadOnlyUsersLDAPRepository.java [hunk 15]
```diff
@@ -634,12 +634,7 @@ public boolean contains(String name) throws UsersRepositoryException {
         return getUserByName(name) != null;
     }
 
-    /*
-     * TODO Should this be deprecated? At least the method isn't declared in the
-     * interface anymore
-     *
-     * @see UsersRepository#containsCaseInsensitive(java.lang.String)
-     */
+    @Deprecated
```
Comment (rouazana) [inline server/data/data-ldap/src/main/java/org/apache/james/user/ldap/ReadOnlyUsersLDAPRepository.java]: idem in this file

File: server/data/data-library/src/main/java/org/apache/james/domainlist/lib/AbstractDomainList.java [hunk 16]
```diff
@@ -210,9 +210,6 @@ public boolean containsDomain(String domain) throws DomainListException {
             .distinct();
     }
 
-    /**
-     * @see #getDomainsIP(List, DNSService, Logger)
-     */
```
Comment (rouazana) [inline server/data/data-library/src/main/java/org/apache/james/domainlist/lib/AbstractDomainList.java]: not an override

File: server/data/data-library/src/main/java/org/apache/james/mailrepository/lib/AbstractMailRepository.java [hunk 17]
```diff
@@ -120,9 +120,6 @@ public void store(Mail mc) throws MessagingException {
         }
     }
 
-    /**
-     * @see #store(Mail)
-     */
```
Comment (rouazana) [inline server/data/data-library/src/main/java/org/apache/james/mailrepository/lib/AbstractMailRepository.java]: not an override

File: server/data/data-library/src/main/java/org/apache/james/mailrepository/lib/AbstractMailRepository.java [hunk 18]
```diff
@@ -151,9 +148,6 @@ public void remove(String key) throws MessagingException {
         }
     }
 
-    /**
-     * @see #remove(String)
-     */
```
Comment (rouazana) [inline server/data/data-library/src/main/java/org/apache/james/mailrepository/lib/AbstractMailRepository.java]: not an override

File: server/mailet/mailetcontainer-camel/src/main/java/org/apache/james/mailetcontainer/impl/jmx/MailProcessorManagement.java [hunk 19]
```diff
@@ -78,9 +78,6 @@ public long getHandledMailCount() {
     /*
      * TODO
      * Is deprecated or dead code? Can't find an interface defining this method
-     * 
-     * @see
-     * org.apache.james.mailetcontainer.api.jmx.MailProcessorDetailMBean#getName()
      */
```
Comment (rouazana) [inline server/mailet/mailetcontainer-camel/src/main/java/org/apache/james/mailetcontainer/impl/jmx/MailProcessorManagement.java]: missing @override?

File: mailbox/jpa/src/main/java/org/apache/james/mailbox/jpa/mail/JPAMessageMapper.java (hunk 4 above)
Comment (chibenwa) [reply mailbox/jpa/src/main/java/org/apache/james/mailbox/jpa/mail/JPAMessageMapper.java]: Nope, this one could be private and does not override something.

Comment (rouazana) [reply mailbox/jpa/src/main/java/org/apache/james/mailbox/jpa/mail/JPAMessageMapper.java]: then you should not remove the comment?

File: server/container/util/src/test/java/org/apache/james/util/retry/naming/NamingExceptionRetryHandlerTest.java (hunk 5 above)
Comment (chibenwa) [reply server/container/util/src/test/java/org/apache/james/util/retry/naming/NamingExceptionRetryHandlerTest.java]: Overrides on setUp()? It's not overriden...

File: protocols/api/src/main/java/org/apache/james/protocols/api/Encryption.java (hunk 10 above)
Comment (chibenwa) [reply protocols/api/src/main/java/org/apache/james/protocols/api/Encryption.java]: @Overides on static methods?

File: protocols/api/src/main/java/org/apache/james/protocols/api/Encryption.java (hunk 11 above)
Comment (chibenwa) [reply protocols/api/src/main/java/org/apache/james/protocols/api/Encryption.java]: Idem

File: protocols/api/src/main/java/org/apache/james/protocols/api/Encryption.java (hunk 10 above)
Comment (rouazana) [reply protocols/api/src/main/java/org/apache/james/protocols/api/Encryption.java]: ok, this comment is useless but unrelated to @override then

File: server/container/util/src/test/java/org/apache/james/util/retry/naming/NamingExceptionRetryHandlerTest.java (hunk 6 above)
Comment (chibenwa) [reply server/container/util/src/test/java/org/apache/james/util/retry/naming/NamingExceptionRetryHandlerTest.java]: Done in a later commit

**Comment by chibenwa** [comment]
test this please

**Comment by mbaechler** [comment]
merged
